maven/bnd-export-maven-plugin/src/main/java/aQute/bnd/maven/export/plugin/ExportMojo.java

```diff
@@ -108,65 +111,46 @@ public int compare(Container o1, Container o2) {
 				}
 			});
 
-			if (!CollectionUtils.isEqualCollection(runBundles, currentRunBundles)) {
-				printRunBundles(runBundles, System.out);
-				if (failOnChanges) {
-					throw new MojoExecutionException("The runbundles have changed. Failing the build");
-				} else {
-					logger.warn("The runbundles have changed");
-					run.setRunBundles(runBundles);
-					File runFile = run.getPropertiesFile();
-					BndEditModel bem = new BndEditModel(run.getWorkspace());
-					Document doc = new Document(IO.collect(runFile));
-					bem.loadFrom(doc);
-
-					List<VersionedClause> bemRunBundles = bem.getRunBundles();
-					List<VersionedClause> deltaAdd = new ArrayList<>();
-					for (Container c : runBundles) {
-						ResourceBuilder rb = new ResourceBuilder();
-						rb.addManifest(Domain.domain(c.getFile()));
-						VersionedClause versionedClause = ResourceUtils.toVersionClause(rb.build(), "[===,==+)");
-						deltaAdd.add(versionedClause);
-					}
-					deltaAdd.removeAll(bemRunBundles);
-					List<VersionedClause> deltaRemove = new ArrayList<>(bemRunBundles);
-					deltaRemove.removeAll(runBundles);
-					boolean added = bemRunBundles.addAll(deltaAdd);
-					boolean removed = bemRunBundles.removeAll(deltaRemove);
-					if (added || removed) {
-						bem.setRunBundles(bemRunBundles);
-						logger.info("Writing changes to {}", runFile.getAbsolutePath());
-						logger.info("{}:{}", Constants.RUNBUNDLES, bem.getDocumentChanges().get(Constants.RUNBUNDLES));
-						bem.saveChangesTo(doc);
-						IO.store(doc.get(), runFile);
-					}
-				}
+			File runFile = run.getPropertiesFile();
+			BndEditModel bem = new BndEditModel(run.getWorkspace());
+			Document doc = new Document(IO.collect(runFile));
+			bem.loadFrom(doc);
+
+			List<VersionedClause> originalBundles = bem.getRunBundles() != null ? bem.getRunBundles()
+					: new ArrayList<VersionedClause>();
+			List<VersionedClause> calculatedRunBundles = new ArrayList<>();
+			for (Container c : runBundles) {
+				ResourceBuilder rb = new ResourceBuilder();
```

Comment:
The way the gradle plugin works avoids the need for Container and ResourceBuilder, so perhaps it is simpler. https://github.com/bndtools/bnd/blob/master/biz.aQute.bnd.gradle/src/aQute/bnd/gradle/BndPlugin.groovy#L480-L491